features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher-new/source/src/app/components/Apis/Details/Configuration/components/StoreVisibility.jsx

```diff
@@ -38,8 +44,43 @@ import Select from '@material-ui/core/Select';
  * @returns
  */
 export default function StoreVisibility(props) {
+    const [roleValidity, setRoleValidity] = useState(true);
     const { api, configDispatcher } = props;
+    const [invalidRoles, setInvalidRoles] = useState([]);
     const isPublic = api.visibility === 'PUBLIC';
+    const handleRoleAddition = (value, role) => {
+        value.then((resp) => {
```

Comment:
Shall we use a meaningful variable name here ?

```suggestion
        value.then((isValid) => {
```

Reply:
Fixed in f20d89587c00c4aadaf582e15f4f469743872571